jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/CaseStageMapper.java

```diff
@@ -32,6 +36,14 @@ public CaseStageSummary apply(CaseStage stage) {
                 .name(stage.getName())
                 .identifier(stage.getIdentifier())
                 .status(stage.getStatus())
+                .adHocFragments(
+                        ofNullable(stage.getAdHocFragments()).orElse(emptyList())
+                                .stream()
+                                .map(new CaseActionAdHocMapper())
+                                .map(s -> {
+                                    s.setStageId(stage.getIdentifier());
```

Comment:
you could pass the stage id in the `CaseActionAdHocMapper` constructor, that avoids calling map twice.